op-node/rollup/derive/calldata_source.go

```diff
@@ -104,11 +113,10 @@ func (ds *DataSource) Next(ctx context.Context) (eth.Data, error) {
 // DataFromEVMTransactions filters all of the transactions and returns the calldata from transactions
 // that are sent to the batch inbox address from the batch sender address.
 // This will return an empty array if no valid transactions are found.
-func DataFromEVMTransactions(config *rollup.Config, batcherAddr common.Address, txs types.Transactions, log log.Logger) []eth.Data {
+func DataFromEVMTransactions(l1Signer types.Signer, batchInboxAddress common.Address, batcherAddr common.Address, txs types.Transactions, log log.Logger) []eth.Data {
```

Comment:
Is there a reason we don't just pass in the `dataSourceConfig` here? Seems like we should either consistently use the config struct or consistently use separate values rather than having a mix of both.

I'd lean towards using the config struct. In that case we should make it an exported type as well since `DataFromEVMTransactions` is exported.

Reply:
Sure, will do. 

While here just saying I'm trying to get my hands on both optimism and go code at the same time so any feedback is welcomed!

Reply:
Should be good @ajsutton, let me know. 

Additionally, although I find the line lengths to be quite long, I am unsure about the formatting conventions used in this project, and I could not find any information regarding formatting standards. Therefore, I have kept the lines as they are.